File: YEAR_3/Application_Development/G-DEV-500-TLS-5-2-area-5/documentation/poc/poc-mobile-kotlin/app/src/main/java/com/area/poc/data/ServicesData.kt
package com.area.poc.data

object ServicesData {
    val availableServices = mapOf(
        "gmail" to Service(
            name = "Gmail",
            icon = "📧",
            actions = listOf(
                ServiceAction(
                    id = "new_email",
                    label = "Nouvel email reçu",
                    description = "Déclenché quand un nouvel email arrive",
                    serviceKey = "gmail"
                ),
                ServiceAction(
                    id = "email_with_attachment",
                    label = "Email avec pièce jointe",
                    description = "Déclenché quand un email avec pièce jointe arrive",
                    serviceKey = "gmail"
                )
            ),
            reactions = emptyList()
        ),
        "onedrive" to Service(
            name = "OneDrive",
            icon = "☁️",
            actions = emptyList(),
            reactions = listOf(
                ServiceReaction(
                    id = "save_file",
                    label = "Sauvegarder un fichier",
                    description = "Sauvegarde un fichier dans OneDrive",
                    serviceKey = "onedrive"
                ),
                ServiceReaction(
                    id = "create_folder",
                    label = "Créer un dossier",
                    description = "Crée un nouveau dossier",
                    serviceKey = "onedrive"
                )
            )
        ),
        "github" to Service(
            name = "GitHub",
            icon = "🐙",
            actions = listOf(
                ServiceAction(
                    id = "new_pr",
                    label = "Nouvelle Pull Request",
                    description = "Déclenché quand une nouvelle PR est créée",
                    serviceKey = "github"
                ),
                ServiceAction(
                    id = "new_issue",
                    label = "Nouvelle Issue",
                    description = "Déclenché quand une nouvelle issue est ouverte",
                    serviceKey = "github"
                )
            ),
            reactions = listOf(
                ServiceReaction(
                    id = "create_issue",
                    label = "Créer une Issue",
                    description = "Crée une nouvelle issue dans un repository",
                    serviceKey = "github"
                ),
                ServiceReaction(
                    id = "star_repo",
                    label = "Starrer un repository",
                    description = "Star un repository spécifié",
                    serviceKey = "github"
                )
            )
        ),
        "twitter" to Service(
            name = "Twitter",
            icon = "🐦",
            actions = listOf(
                ServiceAction(
                    id = "new_tweet",
                    label = "Nouveau tweet",
                    description = "Déclenché quand un nouveau tweet est posté",
                    serviceKey = "twitter"
                ),
                ServiceAction(
                    id = "new_follower",
                    label = "Nouveau follower",
                    description = "Déclenché quand vous gagnez un follower",
                    serviceKey = "twitter"
                )
            ),
            reactions = listOf(
                ServiceReaction(
                    id = "post_tweet",
                    label = "Poster un tweet",
                    description = "Poste un nouveau tweet",
                    serviceKey = "twitter"
                ),
                ServiceReaction(
                    id = "like_tweet",
                    label = "Liker un tweet",
                    description = "Like un tweet spécifié",
                    serviceKey = "twitter"
                )
            )
        )
    )
    
    val allActions: List<ServiceAction> = availableServices.values.flatMap { it.actions }
    val allReactions: List<ServiceReaction> = availableServices.values.flatMap { it.reactions }
}

File: YEAR_3/Application_Development/AREA_first_version/poc/poc-front-vue/src/data/servicesData.js
// Données mockées pour les services
export const availableServices = {
  gmail: {
    name: 'Gmail',
    icon: '📧',
    actions: [
      { id: 'new_email', label: 'Nouvel email reçu', description: 'Déclenché quand un nouvel email arrive' },
      { id: 'email_with_attachment', label: 'Email avec pièce jointe', description: 'Déclenché quand un email avec pièce jointe arrive' }
    ],
    reactions: []
  },
  onedrive: {
    name: 'OneDrive',
    icon: '☁️',
    actions: [],
    reactions: [
      { id: 'save_file', label: 'Sauvegarder un fichier', description: 'Sauvegarde un fichier dans OneDrive' },
      { id: 'create_folder', label: 'Créer un dossier', description: 'Crée un nouveau dossier' }
    ]
  },
  github: {
    name: 'GitHub',
    icon: '🐙',
    actions: [
      { id: 'new_pr', label: 'Nouvelle Pull Request', description: 'Déclenché quand une nouvelle PR est créée' },
      { id: 'new_issue', label: 'Nouvelle Issue', description: 'Déclenché quand une nouvelle issue est ouverte' }
    ],
    reactions: [
      { id: 'create_issue', label: 'Créer une Issue', description: 'Crée une nouvelle issue dans un repository' },
      { id: 'star_repo', label: 'Starrer un repository', description: 'Star un repository spécifié' }
    ]
  },
  twitter: {
    name: 'Twitter',
    icon: '🐦',
    actions: [
      { id: 'new_tweet', label: 'Nouveau tweet', description: 'Déclenché quand un nouveau tweet est posté' },
      { id: 'new_follower', label: 'Nouveau follower', description: 'Déclenché quand vous gagnez un follower' }
    ],
    reactions: [
      { id: 'post_tweet', label: 'Poster un tweet', description: 'Poste un nouveau tweet' },
      { id: 'like_tweet', label: 'Liker un tweet', description: 'Like un tweet spécifié' }
    ]
  }
}

// Helper functions
export const getServiceActions = () => {
  return Object.entries(availableServices).flatMap(([serviceKey, service]) =>
    service.actions.map(action => ({
      service: serviceKey,
      ...action
    }))
  )
}

export const getServiceReactions = () => {
  return Object.values(availableServices).flatMap(service =>
    service.reactions
  )
}